src/solver/opamCudf.ml

```diff
@@ -854,7 +908,10 @@ let extract_explanations packages cudfnv2opam reasons : explanation list =
               Map.update pkg (CS.union chain) CS.empty acc
             in
             let ds = get deps p in
-            let dsc = ds %% all_conflicting in
+            let dsc = match Hashtbl.length missing with
+              | 0 -> ds
+              | _ -> ds %% all_conflicting
+            in
```

Comment:
This feels a bit fragile; maybe we could at least check that there is no "missing" below `p` in the dep tree rather than in general ?

This said, it already avoids the very problematic issue of no messages at all, and thank you *a lot* for finding a workaround.